lib/seattleflu/id3c/cli/command/etl/redcap.py

```diff
@@ -44,3 +69,632 @@ def normalize_net_id(net_id: str=None) -> Optional[str]:
+        if not mapped_symptom:
+            return None
+
+        return { "condition": { "reference": f"#{mapped_symptom}" } }
+
+
+    contained = []
+    diagnosis = []
+
+    for symptom_variable in symptom_onset_map:
+            symptom_keys = []
+
+            for redcap_key in record.keys():
+                symptom_key = grab_symptom_key(record, redcap_key, symptom_variable)
+                if symptom_key:
+                    symptom_keys.append(symptom_key.string)
+
+            symptoms = list(map(lambda x: re.sub('[a-z_]+___', '', x), symptom_keys))
+
+            for symptom in symptoms:
+                contained.append(build_condition(patient_reference, symptom, symptom_onset_map[symptom_variable], system_identifier))
+                diagnosis.append(build_diagnosis(symptom))
+
+    return contained, diagnosis
+
+
+def follow_up_encounter_reason_code() -> dict:
+    encounter_reason_code = create_codeable_concept(
+        system = "http://snomed.info/sct",
+        code = "390906007",
+        display = "Follow-up encounter"
+    )
+    return encounter_reason_code
+
+
+def create_encounter(encounter_date: str, patient_reference: dict, site_reference: dict,
+    collection_code: CollectionCode, record: REDCapRecord, system_identifier: str,
+    locations: list = None, diagnosis: list = None, contained: list = None,
+    parent_encounter_reference: dict = None, encounter_reason_code: dict = None,
+    encounter_identifier_suffix: str = None) -> tuple:
+    """
+    Returns a FHIR Encounter resource entry and reference for the encounter in the study.
+    """
+
+    def build_locations_list(location: dict) -> dict:
+        return {
+            "location": create_reference(
+                reference_type = "Location",
+                reference = location["fullUrl"]
+            )
+        }
+
+
+    def non_tract_locations(resource: dict):
+        return bool(resource) \
+            and resource['resource']['identifier'][0]['system'] != f"{system_identifier}/location/tract"
+
+
+    if not encounter_date:
+        LOG.debug("Not creating the encounter because there is no encounter_date.")
+        return None, None
+
+    if not site_reference:
+        LOG.debug("Not creating the encounter because there is no site_reference.")
+        return None, None
+
+    # Keep the encounter_id format the same as what was used in an earlier
+    # version of redcap_det_uw_reopening.py.
+    if record.event_name:
+        redcap_event_name = record.event_name
+    else:
+        redcap_event_name = ""
+    if record.repeat_instance:
+        redcap_repeat_instance = str(record.repeat_instance)
+    else:
+        redcap_repeat_instance = ""
+    if not encounter_identifier_suffix:
+        encounter_identifier_suffix = ""
+    encounter_id = f"{record.project.base_url}{record.project.id}/{record.id}/{redcap_event_name}/" + \
+        f"{redcap_repeat_instance}{encounter_identifier_suffix}"
```

Comment:
I wonder about keeping this outside of the shared `create_encounter` function and instead have an `encounter_id` parameter that gets passed in. I'm saying this because I was thinking it would be great to start off with hashed `encounter.identifier` for the Childcare study since all of the REDCap metadata is available in `encounter.details`

Reply:
Absolutely. Thanks for catching that. It makes sense for the caller to create the encounter_id and pass it into create_encounter. I've made that change in the "FIXUP: create_encounter : ID as param" commit.